.github/workflows/rust.yml

```
@@ -93,7 +93,7 @@ jobs:
 
       - name: cargo clippy
         working-directory: ./rust
-        run: cargo clippy -- -D warnings
+        run: cargo clippy --all-targets -- -D warnings
```

here we can also add the command to check the code that is hidden behind the feature flags
```suggestion
        run: |
          cargo clippy --all-targets -- -D warnings
          cargo clippy --all-targets --all-features -- -D warnings
```